packages/babel-core/src/transformation/file/index.js

```diff
@@ -52,12 +52,12 @@ export default class File extends Store {
     this.opts = this.initOptions(opts);
 
     this.parserOpts = {
-      highlightCode: this.opts.highlightCode,
-      nonStandard:   this.opts.nonStandard,
-      sourceType:    this.opts.sourceType,
-      filename:      this.opts.filename,
-      sourceFileName:this.opts.filename,
-      plugins:       []
+      allowImportExportEverywhere: this.opts.allowImportExportEverywhere,
+      allowReturnOutsideFunction:  this.opts.allowReturnOutsideFunction,
+      allowSuperOutsideMethod:     this.opts.allowSuperOutsideMethod,
```

Comment:
did you try if you can set this options? I think that the OptionsManager would throw an Error when trying to set these options in e.g `babel.transform(code, { allowImportExportEverywhere: true })` because it does not know the options. So maybe we should only set `sourceType`, `sourceFilename` and `plugins` here because the rest of the options could now be set via the new `parserOpts` option anyway. https://github.com/babel/babel/blob/master/CHANGELOG.md#rocket-new-feature-1